src/datasets/table.py

```diff
@@ -1125,6 +1127,44 @@ def table_cast(table: pa.Table, schema: pa.Schema):
         return table
 
 
+def table_flatten(table: pa.Table):
+    """Improved version of pa.Table.flatten.
+
+    It behaves as pa.Table.flatten, but skips decodable features
+    unless the `decode` attribute of these features is set to False.
```

Comment:
Maybe mention that it does a 1-step flatten to makes things clear (vs Dataset.flatten that flattens everything)